src/core/service.c

```diff
@@ -592,9 +592,15 @@ static int service_verify(Service *s) {
 
 static int service_add_default_dependencies(Service *s) {
         int r;
+        Unit *u;
 
         assert(s);
 
+        u = &s->meta;
+
+        if (u && u->manager && FLAGS_SET(u->manager->test_run_flags, MANAGER_TEST_RUN_IGNORE_DEPENDENCIES))
```

Comment:
Forgot to mention: it's not just the default deps we want to skip when analyzing an image, but all of them. So perhaps it's better to put this check/exit in `unit_add_dependency()` in `src/core/unit.c`? That way all of them are skipped, even the ones that are set manually.